### PR TITLE
SNAP-887 Added wildcard support in snapshot search

### PR DESCRIPTION
**Pull request by pokharelmanish**
### Jira Story

- [Wildcard Characters](https://osi-cwds.atlassian.net/browse/SNAP-887)

## Description
Added wildcard support in snapshot search.

Some examples:
screen shot 2019-02-14 at 10 52 49 am

screen shot 2019-02-14 at 10 52 01 am

screen shot 2019-02-14 at 10 51 31 am

## Tests
- [ ] I have included unit tests 
- [x] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [ ] I have ran lint/rubocop check for the changed/new files.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/helpers/query_builder_helper.rb [hunk 1]
```diff
@@ -12,7 +12,9 @@ module QueryBuilderHelper
   REQUIRE_FIELD_MATCH = 'false'
 
   def formatted_query(string)
-    string.to_s.downcase.gsub(%r{[-/]*(\d+)[-/]*}, '\1')
+    string.to_s.downcase
+          .gsub(%r{[-/]*(\d+)[-/]*}, '\1')
+          .gsub(/[_%]/, '_' => '?', '%' => '*')
```
Comment (pokharelmanish) [inline app/helpers/query_builder_helper.rb]: According to A/C users will enter `_` and `%` for the wildcard search but elastic search support `?` and `*` so I am converting `_ -> ? `and `%-> *`

**Comment by AVandana-OSI** [comment]
retest this please

**Comment by AVandana-OSI** [comment]
retest this please

**Comment by pokharelmanish** [comment]
retest this please

**Pull request merged**